src/libfuncs/felt252.rs

```diff
@@ -253,128 +246,94 @@ pub fn build_binary_operation<'ctx, 'this>(
             let inverse = loop_block.argument(3)?.into();
 
             // First calculate q = rem_(i-1)/rem_i, rounded down
-            let quotient = loop_block
-                .append_operation(arith::divui(prev_remainder, remainder, location))
-                .result(0)?
-                .into();
+            let quotient =
+                loop_block.append_op_result(arith::divui(prev_remainder, remainder, location));
             // Then r_(i+1) = r_(i-1) - q * r_i, and inv_(i+1) = inv_(i-1) - q * inv_i
-            let rem_times_quo = loop_block
-                .append_operation(arith::muli(remainder, quotient, location))
-                .result(0)?
-                .into();
-            let inv_times_quo = loop_block
-                .append_operation(arith::muli(inverse, quotient, location))
-                .result(0)?
-                .into();
-            let next_remainder = loop_block
-                .append_operation(arith::subi(prev_remainder, rem_times_quo, location))
-                .result(0)?
-                .into();
-            let next_inverse = loop_block
-                .append_operation(arith::subi(prev_inverse, inv_times_quo, location))
-                .result(0)?
-                .into();
+            let rem_times_quo =
+                loop_block.append_op_result(arith::muli(remainder, quotient?, location));
+            let inv_times_quo =
+                loop_block.append_op_result(arith::muli(inverse, quotient?, location));
+            let next_remainder =
+                loop_block.append_op_result(arith::subi(prev_remainder, rem_times_quo?, location));
+            let next_inverse =
+                loop_block.append_op_result(arith::subi(prev_inverse, inv_times_quo?, location));
```

Comment:
please use the Result unwraps (the `?`) where you call the method, not where you use the result, otherwise we lose the early return and can make things confusing.

TLDR: 

```rust
 let quotient =
                loop_block.append_op_result(arith::divui(prev_remainder, remainder, location));
```

to

```rust
 let quotient =
                loop_block.append_op_result(arith::divui(prev_remainder, remainder, location))?;
```